static/app/components/links/link.tsx

```diff
@@ -10,42 +10,48 @@ type AnchorProps = React.HTMLProps<HTMLAnchorElement>;
 type ToLocationFunction = (location: Location) => LocationDescriptor;
 
 type Props = WithRouterProps & {
-  // URL
+  /**
+   * The string path or LocationDescriptor object
+   */
   to: ToLocationFunction | LocationDescriptor;
-  // Styles applied to the component's root
+  /**
+   * Style applied to the component's root
+   */
   className?: string;
 } & Omit<AnchorProps, 'href' | 'target' | 'as' | 'css'>;
 
 /**
  * A context-aware version of Link (from react-router) that falls
  * back to <a> if there is no router present
  */
-class Link extends React.Component<Props> {
-  componentDidMount() {
-    const isRouterPresent = this.props.location;
-    if (!isRouterPresent) {
+const BaseLink: React.FC<Props> = ({location, disabled, to, ref, ...props}) => {
+  useEffect(() => {
+    // check if the router is present
+    if (!location) {
       Sentry.captureException(
         new Error('The link component was rendered without being wrapped by a <Router />')
       );
     }
+  }, [location]);
+
+  if (!disabled && location) {
+    return <RouterLink to={to} ref={ref as any} {...props} />;
   }
 
-  render() {
-    const {disabled, to, ref, location, ...props} = this.props;
+  if (typeof to === 'string') {
+    return <Anchor href={to} ref={ref} disabled={disabled} {...props} />;
+  }
 
-    if (!disabled && location) {
-      return <RouterLink to={to} ref={ref as any} {...props} />;
-    }
+  return <Anchor href="" ref={ref} {...props} disabled />;
+};
 
-    if (typeof to === 'string') {
-      return <Anchor href={to} ref={ref} disabled={disabled} {...props} />;
-    }
+// Set the displayName for testing convinience
+BaseLink.displayName = 'Link';
 
-    return <Anchor href="" ref={ref} {...props} disabled />;
-  }
-}
+// Re-assign to Link to make auto-importing smarter
+const Link = withRouter(BaseLink);
```

Comment:
it would be super nice if we could use hooks here, so we wouldn't have this problem :/

Reply:
React router 6 is only hooks, so your wish will come true soon :)